### PR TITLE
Enable Codespaces for ISPC

### PR DESCRIPTION
**Pull request by dbabokin**
PR includes a minimal Codespaces container for comfortable browsing, editing, and building ISPC project.

The configuration includes pre-built LLVM 15.0.7.

If it works well, we can add GPU-enabled env, though I'm not sure if we can have Intel GPU-enabled cloud instance.

### CODE REVIEW

File: .devcontainer/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+To build ISPC and run the tests do the following:
+```bash
+# `build` folder already contains result of `cmake .. -DCMAKE_EXPORT_COMPILE_COMMANDS=ON`
+cd build
+# build ISPC and examples
+make -j4
+# run LIT tests
+make check-all
+```
```
Comment (e3m3) [inline .devcontainer/README.md:14]: The contents of the root directory does not contain `build`:
```shell
@e3m3 ➜ /workspaces/ispc (main) $ ls
CMakeLists.txt   alloy.py            check_env.py     common.py    examples      perf.py       stdlib2cpp.py       tests_errors
CONTRIBUTING.md  benchmarks          check_format.sh  contrib      fail_db.txt   run_tests.py  test_static.cpp     third-party-programs.txt
LICENSE.txt      bitcode2cpp.py      check_isa.cpp    docker       ispcrt        scripts       test_static.isph    utils
README.md        builtins            cmake            docs         llvm_patches  src           test_static_l0.cpp
SECURITY.md      check_copyright.sh  common           doxygen.cfg  perf.ini      stdlib.ispc   tests
```

Comment (dbabokin) [reply .devcontainer/README.md:14]: Check you build logs, container build probably failed. Before it's merged you need to manually set my repo and my branch as a source.

**Review by nurmukhametov — COMMENTED** [review]
I have tested that it works well. Will it be more useful to ship (LLVM) and pre-configure debug build (ISPC)?

**Review by aneshlya — APPROVED** [review]

**Comment by dbabokin** [comment]
> I have tested that it works well. Will it be more useful to ship (LLVM) and pre-configure debug build (ISPC)?

Hmm... This does make sense (when it comes to ISPC build). Shipping debug LLVM build is bad idea and as I understand you, you are not proposing that.

From another prospective, the key use case for Codespaces is to hack the code, test it and commit, not to debug. And building Release is better for that.

I suggest going forward "as is", see how it works and add more containers/configurations when we see that the base scenario works well.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "__comment1__": "Copyright (c) 2023, Intel Corporation",
+  "__comment2__": "SPDX-License-Identifier: BSD-3-Clause",
+  "name": "ISPC developers environment (CPU only)",
+  "image": "mcr.microsoft.com/devcontainers/universal:2",
+  "features": {
+    "ghcr.io/devcontainers-contrib/features/apt-get-packages:1": { "packages" : "libc6-dev,libc6-dev-i386,flex,wget" },
```
[the remark this answers is not in the document]
Comment (dbabokin) [reply .devcontainer/devcontainer.json:7]: Ok, as `libtbb-dev` is not settled yet, I'll let you add when it, when you are settled on the specific way of adding it.

**Pull request merged**